packages/synapse-sdk/src/test/synapse.test.ts

```diff
@@ -946,6 +955,86 @@ describe('Synapse', () => {
       assert.isTrue(defaultContexts === contexts)
     })
 
+    it('prefers to select at least one context with an endorsement', async function () {
+      this.timeout(15000)
+      const providerId1 = Number(PROVIDERS.provider1.providerId)
+      const providerId2 = Number(PROVIDERS.provider2.providerId)
+      // verify that without an endorsement distinction, this function selects randomly
+      const counts = {
+        [providerId1]: 0,
+        [providerId2]: 0,
+      }
+      for (let i = 0; i < 14; i++) {
+        const contexts = await synapse.storage.createContexts({
+          forceCreateDataSets: true, // This prevents the defaultContexts caching
+        })
+        assert.equal(contexts.length, 2)
+        assert.notEqual(contexts[0].provider.id, contexts[1].provider.id)
+        assert.equal((contexts[0] as any)._dataSetId, undefined)
+        assert.equal((contexts[1] as any)._dataSetId, undefined)
+        counts[contexts[0].provider.id]++
+      }
+      // These tests can fail probabilistically
+      assert.isAbove(counts[Number(PROVIDERS.provider1.providerId)], 0)
+      assert.isAbove(counts[Number(PROVIDERS.provider2.providerId)], 0)
```

Comment:
I'd like to verify this assumption in a faster and more reliable way, (such as #390). Without this step though, the subsequent test could someday pass even if the endorsement preference was broken.